src/cdk/tree/nested-node.ts

```diff
@@ -72,14 +72,24 @@ export class CdkNestedTreeNode<T, K = T>
     if (!this._tree.treeControl.getChildren && (typeof ngDevMode === 'undefined' || ngDevMode)) {
       throw getTreeControlFunctionsMissingError();
     }
-    const childrenNodes = this._tree.treeControl.getChildren(this.data);
-    if (Array.isArray(childrenNodes)) {
-      this.updateChildrenNodes(childrenNodes as T[]);
-    } else if (isObservable(childrenNodes)) {
-      childrenNodes
-        .pipe(takeUntil(this._destroyed))
-        .subscribe(result => this.updateChildrenNodes(result));
-    }
+
+    this._dataChanges
+      .pipe(
+        takeUntil(this._destroyed),
```

Comment:
We typically put the `takeUntil(this._destroyed)` at the end of the chain to avoid accidental leak (there's an article about it here: https://ncjamieson.com/avoiding-takeuntil-leaks/)

Reply:
Interesting, thanks! Fixed it.